tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -1498,8 +1571,10 @@ private[emitter] class JSDesugaring(semantics: Semantics,
           } else {
             val names = fields map (_._1)
             val items = fields map (_._2)
-            unnest(items) { (newItems, env) =>
-              redo(JSObjectConstr(names.zip(newItems)))(env)
+            unnestPropertyNames(names) { (newNames, env) =>
+              unnest(items) { (newItems, env) =>
```

Comment:
Experimentation in Firefox and reading the spec confirm that the evaluation order is interleaved between keys and values. So this is not correct.